feature_engine/datetime/datetime.py

```diff
@@ -159,7 +159,7 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
 
         if self.features_to_extract is None:
             self.features_to_extract_ = FEATURES_DEFAULT
-        elif self.features_to_extract == "all":
+        elif isinstance(self.features_to_extract, str):
```

Comment:
Based on [this](https://stackoverflow.com/questions/51656300/mypy-falsely-reports-error-on-union-typed-variable-when-branching-on-type)
It should be harmless since we reject any string that isn't equal to _all_ in a previous check